main.go

```diff
@@ -75,8 +81,42 @@ func main() {
 	flagSet.StringVar(&leaderElectResourceNamespace, "leader-elect-resource-namespace", "openshift-cluster-machine-approver", "the namespace in which the leader election resource will be created.")
 	flagSet.Parse(os.Args[1:])
 
-	if err := validateAPIGroup(APIGroup); err != nil {
-		klog.Fatalf(err.Error())
+	// Deprecated options
+	flagSet.StringVar(&apiGroup, "apigroup", "machine.openshift.io", "API group for machines")
```

Comment:
This should now have an empty value else you can't set the new flag
```suggestion
	flagSet.StringVar(&apiGroup, "apigroup", "", "API group for machines")
```